### PR TITLE
Divider component

### PR DESCRIPTION
**Pull request by dawinValenzuela**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://impulsumvc.atlassian.net/browse/COM-210

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** (Link to a local storybook url to review new functionality)

### CODE REVIEW

File: packages/components/src/components/primitives/Divider/Divider.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import TestRendered from 'react-test-renderer';
+import { Divider } from './Divider';
+
+describe('Divider', () => {
+  it('Should render', () => {
+    const { getByTestId } = renderWithProviders(
+      <Divider data-testid="divider-id" />
+    );
+
+    getByTestId('divider-id');
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Divider/Divider.test.tsx:12]: use `screen.getByTestId` instead, see https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

**Review by tkanzakic — COMMENTED** [review]
small comment and LGTM

File: packages/components/src/components/primitives/Divider/Divider.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  <VStack divider={<StackDivider borderColor="gray.200" />} spacing={4}>
+    {DividerOrientations.map((orientation, id) => (
+      <Template orientation={orientation} key={id} />
+    ))}
+  </VStack>
+);
+export const All = AllTemplate.bind({});
+
+const Template = (args) => (
+  <Center height="50px" width="full">
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Divider/Divider.stories.tsx:28]: why not just render  the divider?

File: packages/components/src/components/primitives/Divider/Divider.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  argTypes: {
+    orientation: {
+      options: DividerOrientations,
+      control: { type: 'select' },
+    },
+  },
+} as Meta;
+
+const AllTemplate = () => (
+  <VStack divider={<StackDivider borderColor="gray.200" />} spacing={4}>
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Divider/Divider.stories.tsx:19]: this divider makes hard to identify the actual divider component

File: packages/components/src/components/primitives/Divider/Divider.test.tsx (hunk 1 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Divider/Divider.test.tsx]: done!.

File: packages/components/src/components/primitives/Divider/Divider.stories.tsx (hunk 2 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Divider/Divider.stories.tsx]: when the orientation change to vertical, the parent needs to have a height or I'm gonna change it and add the height to the Divider.

File: packages/components/src/components/primitives/Divider/Divider.stories.tsx (hunk 3 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Divider/Divider.stories.tsx]: removed!.

**Review by tkanzakic — APPROVED** [review]
LGTM!

**Pull request merged**